Mlem/Views/Shared/Components/Modlog/ModlogView.swift

```diff
@@ -0,0 +1,375 @@
+//
+//  ModlogView.swift
+//  Mlem
+//
+//  Created by Eric Andrews on 2024-03-10.
+//
+
+import Dependencies
+import Foundation
+import SwiftUI
+
+enum ModlogAction: CaseIterable {
+    case all, postRemoval, postLock, postPin, commentremoval, communityRemoval, communityBan, instanceBan,
+         moderatorAdd, communityTransfer, administratorAdd, personPurge, communityPurge, postPurge, commentPurge, communityHide
+    
+    var label: String {
+        switch self {
+        case .all:
+            "All"
+        case .postRemoval:
+            "Removed Post"
+        case .postLock:
+            "Locked Post"
+        case .postPin:
+            "Pinned Post"
+        case .commentremoval:
+            "Removed Comment"
+        case .communityRemoval:
+            "Removed Community"
+        case .communityBan:
+            "Banned from Community"
+        case .instanceBan:
+            "Banned from Instance"
+        case .moderatorAdd:
+            "Appointed Moderator"
+        case .communityTransfer:
+            "Transferred Community"
+        case .administratorAdd:
+            "Appointed Administrator"
+        case .personPurge:
+            "Purged Person"
+        case .communityPurge:
+            "Purged Community"
+        case .postPurge:
+            "Purged Post"
+        case .commentPurge:
+            "Purged Comment"
+        case .communityHide:
+            "Hid Community"
+        }
+    }
+}
+
+// swiftlint:disable:next type_body_length
+struct ModlogView: View {
+    @Dependency(\.apiClient) var apiClient
+    @Dependency(\.errorHandler) var errorHandler
+    
+    // TODO: 2.0 enable searching--search needs to be submitted against the instance that the modlog is fetched from to ensure that the communityId/moderatorId is locally correct, which is annoying right now but very easy in 2.0.
+    
+    @State var selectedAction: ModlogAction = .all
+    
+    @State var currentTracker: any ModlogTrackerProtocol
+    @StateObject var modlogTracker: ModlogTracker
+    @StateObject var postRemovalsTracker: ModlogChildTracker
+    @StateObject var postLocksTracker: ModlogChildTracker
+    @StateObject var postPinsTracker: ModlogChildTracker
+    @StateObject var commentRemovalsTracker: ModlogChildTracker
+    @StateObject var communityRemovalsTracker: ModlogChildTracker
+    @StateObject var communityBansTracker: ModlogChildTracker
+    @StateObject var instanceBansTracker: ModlogChildTracker
+    @StateObject var moderatorAddsTracker: ModlogChildTracker
+    @StateObject var communityTransfersTracker: ModlogChildTracker
+    @StateObject var administratorAddsTracker: ModlogChildTracker
+    @StateObject var personPurgesTracker: ModlogChildTracker
+    @StateObject var communityPurgesTracker: ModlogChildTracker
+    @StateObject var postPurgesTracker: ModlogChildTracker
+    @StateObject var commentPurgesTracker: ModlogChildTracker
+    @StateObject var communityHidesTracker: ModlogChildTracker
```

Comment:
It might be better to put these in a dictionary mapping `ModLogAction` to `ModlogChildTracker`? That means we can initialise these trackers iteratively, and replace the switch statement on line 248 with a simple dictionary access

Reply:
On second thought, this probably won't work without `@Observable`. Probably doable in 2.0 though